core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorTableExecuteHandle.java

```diff
@@ -23,4 +23,7 @@
  * {@link ConnectorMetadata#beginTableExecute(ConnectorSession, ConnectorTableExecuteHandle, ConnectorTableHandle)}
  * The tableHandle passed to beginTableExecute is one obtained from matching TableScanNode at the of planning.
  */
-public interface ConnectorTableExecuteHandle {}
+public interface ConnectorTableExecuteHandle
+{
+    boolean isReadsData();
```

Comment:
Handle should be opaque. 
This belongs to `TableProcedureMetadata`

Reply:
as discussed offline, TableProcedureMetadata + Analysis to pass the info between statement analyzer  and the planner.